sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BulkOperations.java

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.batch.ItemBulkOperation;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.util.Beta;
+import reactor.core.publisher.Flux;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Utility for creating bulk operations which can be executed by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)} .
+ *
+ * Also while creating these operation, if some options which are only for individual operation can be provided by passing
+ * a {@link BulkItemRequestOptions} while creating the bulk operation.
+ *
+ * See also {@link BulkProcessingOptions}.
+ */
+@Beta(Beta.SinceVersion.V4_9_0)
```

Comment:
@milismsft @kushagraThapar should we keep Beta tag on class or all it's functions? What's the best way?

Reply:
Discussed with Milis. Beta tag on class makes more sense for newly added class. Also later, if a function is added to in class which is in beta, that function will have new beta.

Reply:
absolutely :-)